### PR TITLE
Crypto

### PR DESCRIPTION
**Pull request by cheng-tan**

### CODE REVIEW

File: ios/tracedefense1Tests/Encryption/EncryptionUnitTest.swift [hunk 1]
```diff
@@ -0,0 +1,38 @@
+//
+//  EncryptionUnitTest.swift
+//  tracedefense1Tests
+//
+//  Created by Cheng Tan on 4/17/20.
+//  Copyright © 2020 Facebook. All rights reserved.
```
Comment (kumpera) [inline ios/tracedefense1Tests/Encryption/EncryptionUnitTest.swift]: ??

File: ios/Encryption/EncryptionUtil.swift [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+            let hmacKey = try KeyChain.get(tag: HMAC_KEY_TAG) ?? createKey(tag: HMAC_KEY_TAG)
+            print("HMAC key: \(hmacKey.base64EncodedString())")
+            let hmac = try HMAC(key: hmacKey)
+            let hmacVal: Data = try hmac.getHMAC(encryptedData)
+            encryptedData.append(hmacVal)
+            print(encryptedData)
+            return encryptedData.base64EncodedString()
+        } catch {
+          print("Error: \(error)")
```
Comment (ApolloZhu) [inline ios/Encryption/EncryptionUtil.swift]: The project is configured to use 2 spaces for indentation. Could you please reformat the files?

File: ios/Encryption/EncryptionUtil.swift [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    static let KEY_SIZE = kCCKeySizeAES256
+    static let HMAC_SIZE = Int(CC_SHA256_DIGEST_LENGTH)
+    
+    static let AES_KEY_TAG = "covidsafe.keys.aes_key"
+    static let HMAC_KEY_TAG = "covidsafe.keys.hmac_key"
+
+    static func encrypt(plainText: String) -> String {
+        do {
+            let aesKey = try KeyChain.get(tag: AES_KEY_TAG) ?? createKey(tag: AES_KEY_TAG)
+            print("AES key: \(aesKey.base64EncodedString())")
```
Comment (ApolloZhu) [inline ios/Encryption/EncryptionUtil.swift]: While knowing this during debugging is useful, should we really print it out during production?

File: ios/Encryption/AES.swift [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+        self.key = key
+    }
+}
+
+extension AES {
+    func encrypt(_ plainText: String) throws -> Data {
+        let dataToEncrypt = Data(plainText.utf8)
+        let bufferSize: Int = ivSize + dataToEncrypt.count + kCCBlockSizeAES128 // allow result round up to up the nearest multiple of 16 bytes
+        var buffer = Data(count: bufferSize)
+        try EncryptionUtil.randomData(length: ivSize, for: &buffer) //generate iv
```
Comment (ApolloZhu) [inline ios/Encryption/AES.swift]: Just for consistency, could you please add a whitespace after `//` like all your other comments?

**Pull request merged**